packages/server/src/unstable-core-do-not-import/http/resolveResponse.ts

```diff
@@ -371,9 +379,27 @@ export async function resolveResponse<TRouter extends AnyRouter>(
 
     // ----------- response handlers -----------
     if (!info.isBatchCall) {
-      const [call] = info.calls;
+      const call = info.calls[0]!;
       const [error, result] = await rpcCalls[0]!;
 
+      if (!error && result?.data instanceof Response) {
+        const response = result.data;
+        if (!config.experimental?.outputResponse) {
+          throw new Error(
+            'You need to activate outputResponse in the root config to use this feature - e.g. `initTRPC.create({ experimental: { outputResponse: true } })`',
+          );
+        }
+
+        const headers = new Headers(response.headers);
+        headers.set('trpc-response-output', '1');
+
+        return new Response(response.body, {
```

Comment:
This seems more inspired by jstack than tRPC

tRPC would say this is a `.output()` call. I don't know if we want to be exposing http inside procedures really? One of the cool things is we/others could build adapters and links which run over entirely different protocols and this would get in the way of that.

Reply:
Advantage of an output() call is in theory that could be checked for some metadata before the procedure ever gets executed to complain about being part of a batched call

```ts
const blobOutput = {
  parse(),
  trpc: { supportsBatching: false }
}

procedure.output(blobOutput).query()
```

Reply:
i'm unsure what you're suggesting here and how it would look like - what are you saying a procedure should return in this case instead?

Reply:
Updated my code example above